src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -19,73 +19,154 @@
 
 package net.minecraftforge.items;
 
+import com.google.common.collect.Range;
 import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.filter.IStackFilter;
 
 import javax.annotation.Nonnull;
+import java.util.OptionalInt;
 
 public interface IItemHandler
 {
+
+    /**
+     * @return the size of this IItemHandler
+     */
+    int size();
+
+    /**
+     * @param stack to check
+     * @return true is the stack can be inserted
+     */
+    default boolean isStackValid(@Nonnull ItemStack stack)
+    {
+        for (int i = 0; i < size(); i++)
+        {
+            if (isStackValidForSlot(stack, i))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param stack to check
+     * @return true is the stack can be extracted
+     */
+    default boolean canExtractStack(@Nonnull ItemStack stack)
+    {
+        for (int i = 0; i < size(); i++)
+        {
+            if (canExtractStackFromSlot(stack, i))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param stack to check
+     * @param slot  to check
+     * @return true is the stack can be inserted into the slot
+     */
+    default boolean isStackValidForSlot(@Nonnull ItemStack stack, int slot)
```

Comment:
reverse the arguments; convention has always been slot args first when they are present